addons/binding/org.openhab.binding.milight/src/main/java/org/openhab/binding/milight/internal/protocol/QueuedSend.java

```diff
@@ -0,0 +1,276 @@
+ *
+ */
+public class QueuedSend implements Runnable {
+    private static class QueueItem {
+        byte[] data;
+        int unique_command_id;
+        boolean repeatable;
+        int custom_delay_time;
+
+        public QueueItem(byte[] data, int unique_command_id, boolean repeatable, int custom_delay_time) {
+            super();
+            this.data = data;
+            this.unique_command_id = unique_command_id;
+            this.repeatable = repeatable;
+            this.custom_delay_time = custom_delay_time;
+        }
+    }
+
+    BlockingQueue<QueueItem> queue = new LinkedBlockingQueue<>(20);
+    protected final DatagramPacket packet;
+    protected final DatagramSocket datagramSocket;
+    private static final Logger logger = LoggerFactory.getLogger(QueuedSend.class);
+    private int delay_between_commands = 100;
+    private int repeat_commands = 1;
+    private boolean willbeclosed = false;
+    private Thread thread;
+
+    public static final byte NO_CATEGORY = 0;
+
+    /**
+     * Creates a new send queue and starts the background thread.
+     *
+     * @param addr May be null but has to be set by setAddress before first call to queue().
+     * @param port The destination port
+     * @throws SocketException
+     */
+    public QueuedSend(InetAddress addr, int port) throws SocketException {
+        byte[] a = new byte[0];
+        packet = new DatagramPacket(a, a.length, addr, port);
+        datagramSocket = new DatagramSocket();
+        thread = new Thread(this);
+        thread.start();
+    }
+
+    public int getDelayBetweenCommands() {
+        return delay_between_commands;
+    }
+
+    public int getRepeatCommands() {
+        return repeat_commands;
+    }
+
+    public void setRepeatCommands(int repeat_commands) {
+        repeat_commands = Math.max(0, Math.min(5, repeat_commands));
+        this.repeat_commands = repeat_commands;
+    }
+
+    public void setDelayBetweenCommands(int ms) {
+        ms = Math.max(0, Math.min(400, ms));
+        delay_between_commands = ms;
+    }
+
+    /**
+     * The queue process
+     */
+    @Override
+    public void run() {
+        while (!willbeclosed) {
+            QueueItem item;
+            int delay_time = delay_between_commands;
+            try {
+                // block/wait for another item
+                item = queue.take();
+            } catch (InterruptedException e) {
+                if (!willbeclosed) {
+                    logger.error("Queue take failed: " + e.getLocalizedMessage());
+                }
+                break;
+            }
+
+            if (item == null) {
+                continue;
+            }
+
+            if (item.custom_delay_time != 0) {
+                delay_time = item.custom_delay_time;
+            }
+
+            packet.setData(item.data);
+            try {
+                int repeat_remaining = item.repeatable ? repeat_commands : 1;
+                repeat_remaining = Math.min(repeat_remaining, 1);
```

Comment:
I don't think the repeats work since this code looks like it will always set the number of repeats to 1:

repeat_remaining = Math.min(repeat_remaining, 1);